### PR TITLE
SNOW-1437295 ensure unrecommended JDBC driver must be forced

### PR DESCRIPTION
**Pull request by sfc-gh-mhofman**
Support for detecting unrecommended JDBC Driver and toggle to force it.

### CODE REVIEW

File: snowflake-hibernate/src/main/java/net/snowflake/hibernate/dialect/SnowflakeDialect.java [hunk 1]
```diff
@@ -28,9 +26,9 @@ public class SnowflakeDialect extends Dialect {
     private final static String CONFIGURATION_PROPERTY_PREFIX = "hibernate.dialect.snowflake.";
     final static String CONFIGURATION_PROPERTY_TABLE_TYPE = CONFIGURATION_PROPERTY_PREFIX + "table_type";
     final static String CONFIGURATION_PROPERTY_DEVELOPMENT_MODE = CONFIGURATION_PROPERTY_PREFIX + "development_mode";
+    final static String ALLOW_UNRECOMMENDED_JDBC_DRIVER_VERSION = CONFIGURATION_PROPERTY_PREFIX + "hibernate.dialect.snowflake.allow_unrecommended_jdbc_driver";
```
Comment (sfc-gh-pbulawa) [inline snowflake-hibernate/src/main/java/net/snowflake/hibernate/dialect/SnowflakeDialect.java:29]: No need to add `hibernate.dialect.snowflake` as it is the prefix.

**Review by sfc-gh-pbulawa — CHANGES_REQUESTED** [review]

File: snowflake-hibernate/src/main/java/net/snowflake/hibernate/dialect/SnowflakeDialect.java [hunk 2]
```diff
@@ -47,16 +45,25 @@ public SnowflakeDialect(DialectResolutionInfo info) {
...
+        warnOnUnsupportedDriverVersion(allowUnrecommendedJdbcVersion);
     }
 
-    private void warnOnUnsupportedDriverVersion() {
+    private void warnOnUnsupportedDriverVersion(boolean allowUnrecommendedJdbcDriverVersion) {
         // security risk mitigation
         // we cannot use info.getDriverMajorVersion() and info.getDriverMinorVersion() since there is no method for patch part of the version
         try {
             Version driverVersion = Version.from(SnowflakeDriver.implementVersion);
+            log.info("Dialect property {} has value {}", ALLOW_UNRECOMMENDED_JDBC_DRIVER_VERSION, allowUnrecommendedJdbcDriverVersion);
```
Comment (sfc-gh-dprzybysz) [inline snowflake-hibernate/src/main/java/net/snowflake/hibernate/dialect/SnowflakeDialect.java:57]: we can set debug or trace for this log

**Review by sfc-gh-dprzybysz — CHANGES_REQUESTED** [review]

File: snowflake-hibernate/src/main/java/net/snowflake/hibernate/dialect/SnowflakeDialect.java (hunk 1 above)
Comment (sfc-gh-mhofman) [reply snowflake-hibernate/src/main/java/net/snowflake/hibernate/dialect/SnowflakeDialect.java:29]: Right, I fixed that in another pr, but let's have it valid here as well

File: snowflake-hibernate/src/main/java/net/snowflake/hibernate/dialect/SnowflakeDialect.java (hunk 2 above)
Comment (sfc-gh-mhofman) [reply snowflake-hibernate/src/main/java/net/snowflake/hibernate/dialect/SnowflakeDialect.java:57]: I'll remove it - there is logDialectProperties method already

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Review by sfc-gh-pbulawa — APPROVED** [review]

**Pull request merged**